packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -298,15 +312,39 @@ class _CodeViewState extends State<CodeView>
                         Expanded(
                           child: LayoutBuilder(
                             builder: (context, constraints) {
-                              return Lines(
-                                constraints: constraints,
-                                scrollController: textController,
-                                lines: lines,
-                                pausedFrame: pausedFrame,
-                                searchMatchesNotifier:
-                                    widget.controller.searchMatches,
-                                activeSearchMatchNotifier:
-                                    widget.controller.activeSearchMatch,
+                              final num longestLineLength = lines.fold(
+                                  0,
+                                  (longestLength, line) => math.max(
+                                      longestLength,
+                                      line.toPlainText().length));
+                              final fileWidth =
+                                  CodeView.assumedCodeCharacterWidth *
+                                      longestLineLength;
+                              final boxWidth = math.max(
+                                  constraints.minWidth, constraints.maxWidth);
+
+                              return Scrollbar(
+                                isAlwaysShown: fileWidth > boxWidth,
+                                controller: horizontalController,
+                                child: SingleChildScrollView(
+                                  scrollDirection: Axis.horizontal,
+                                  controller: horizontalController,
+                                  child: Lines(
+                                    constraints: BoxConstraints(
+                                      maxHeight: constraints.maxHeight,
+                                      minHeight: constraints.minHeight,
+                                      minWidth: math.max(fileWidth, boxWidth),
```

Comment:
why not constrain to exactly the fileWidth?

Reply:
Good point, now constraining to just `fileWidth`